src/index.ts

```diff
@@ -57,51 +59,59 @@ export function koaMiddleware<TContext extends BaseContext>(
       return;
     }
 
-    const headers = new Map<string, string>();
-    for (const [key, value] of Object.entries(ctx.headers)) {
-      if (value !== undefined) {
-        // Node/Koa headers can be an array or a single value. We join
-        // multi-valued headers with `, ` just like the Fetch API's `Headers`
-        // does. We assume that keys are already lower-cased (as per the Node
-        // docs on IncomingMessage.headers) and so we don't bother to lower-case
-        // them or combine across multiple keys that would lower-case to the
-        // same value.
-        headers.set(
-          key,
-          Array.isArray(value) ? value.join(', ') : (value as string),
-        );
+    const incomingHeaders = new Map(function*() {
```

Comment:
Is this change necessary or beneficial? I only see additional complexity by introducing an II(generator)FE.

Reply:
It's not necessary, just a personal style thing. I was changing the name of this variable and the rest was reflex. I use this IIFE style a lot because it means you don't need to specify types [`new Map` vs `new Map<string, string>`], you don't need to repeat the name of the container [`yield` vs `incomingHeaders.set(...)`], and if a temporary variable is needed then it is contained to a nested scope. This also mirrors [do expressions](https://github.com/tc39/proposal-do-expressions) which will hopefully make it into the language in one way or another. Especially with TypeScript this style is super helpful. Anyway I reverted the change because it's not important.